src/Compilers/CSharp/Portable/Lowering/AsyncRewriter/AsyncIteratorMethodToStateMachineRewriter.cs

```diff
@@ -143,6 +153,9 @@ protected override BoundStatement GenerateSetExceptionCall(LocalSymbol exception
             // if (this.combinedTokens != null) { this.combinedTokens.Dispose(); this.combinedTokens = null; } // for enumerables only
             AddDisposeCombinedTokensIfNeeded(builder);
 
+            // _current = default;
+            builder.Add(GenerateClearCurrent());
```

Comment:
>builder.Add(GenerateClearCurrent()); [](http://example.com/codeflow?start=12&length=36)

Similar question about the order of evaluation.

Reply:
I think the order is okay here too (we clear `Current` after disposing `combinedTokens` and before calling method on `_promiseOfValueOrEnd`, but OP description is indeed missing call to `builder.Complete()`. I'll need to update the notes.
In terms of order, the important thing is to clear `Current` before calling `SetException`/`SetResult` on `_promiseOfValueOrEnd`.

Reply:
Updated OP to reflect. Thanks